cocotb/__init__.py

```diff
@@ -95,13 +96,37 @@ def _reopen_stream_with_buffering(stream_name):
 # so that cocotb.scheduler gives you the singleton instance and not the
 # scheduler package
 
-scheduler = None
+scheduler = None  # type: cocotb.scheduler.Scheduler
 """The global scheduler instance."""
 
-regression_manager = None
+regression_manager = None  # type: cocotb.regression.RegressionManager
+"""The global regression manager instance."""
 
-plusargs = {}
-"""A dictionary of "plusargs" handed to the simulation."""
+argv = None  # type: List[str]
+"""The argument list as seen by the simulator"""
+
+argc = None  # type: int
+"""The length of :data:`cocotb.argv`"""
+
+PLUSARGS = None  # type: Dict[str, Union[bool, str]]
+"""A dictionary of "plusargs" handed to the simulation. See :make:var:`PLUSARGS` for details."""
+
+plusargs = PLUSARGS  # preserves legacy naming
```

Comment:
How does this work? Updating one won't update the other.

Reply:
It used to be a dict, but yeah it definitely won't now, will fix.